### PR TITLE
As a developer I want to have clear tests structure and example (Falcon)

### PR DESCRIPTION
**Pull request by kalabro**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -33,12 +33,23 @@ jobs:
...
+          name: Prepare tests
+          command: make tests-prepare
 
       - run:
-          name: Check drush status
-          command: make drush status
+          name: Run unit (integration) tests
+          command: |
+            make tests-run unit || true
+            make tests-run-failed unit
```
Comment (spleshka) [inline .circleci/config.yml]: Do we allow for unit tests to fail? I'm not sure if this is possible to have false positives for units.

File: .circleci/config.yml [hunk 2]
```diff
@@ -33,12 +33,23 @@ jobs:
...
+
+      - run:
+          name: Run API tests
+          command: |
+            make tests-run api || true
+            make tests-run-failed api
```
Comment (spleshka) [inline .circleci/config.yml]: Do we allow for api tests to fail? I'm not sure if this is possible to have false positives for api.

File: Makefile [hunk 3]
```diff
@@ -92,6 +92,32 @@ ifeq ($(ENV), development)
...
+	rm -rf .codecept/.git
+
+tests-run:
+	$(call message,$(PROJECT_NAME): Run Codeception tests)
+	$(eval ARGS := $(filter-out $@,$(MAKECMDGOALS)))
+	$(eval TARGET := $(firstword $(ARGS)))
+	docker-compose run --rm codecept run $(TARGET) --debug
+
+tests-run-failed:
+	$(call message,$(PROJECT_NAME): Run Codeception tests)
```
Comment (spleshka) [inline Makefile]: Run `Failed` Codeception tests.

File: tests/api.suite.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+actor: ApiTester
```
Comment (spleshka) [inline tests/api.suite.yml:1]: Can we have 2 spaces intent here?

File: tests/unit.suite.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Codeception Test Suite Configuration
```
Comment (spleshka) [inline tests/unit.suite.yml:1]: Can we have 2 spaces intent here?

File: tests/unit/falcon_development/DevelopmentTest.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                \Drupal::moduleHandler()->moduleExists($module),
+                "Module $module should be enabled."
+            );
+        }
+    }
+
+    /**
+     * Checks that test users esist and enabled.
+     */
+    public function testTestUsersEnabled()
```
Comment (spleshka) [inline tests/unit/falcon_development/DevelopmentTest.php]: I'm soooo excited !

File: .circleci/config.yml (hunk 2 above)
Comment (kalabro) [reply .circleci/config.yml]: Removing for now. We can enable them later.

File: tests/unit/falcon_development/DevelopmentTest.php (hunk 6 above)
Comment (kalabro) [reply tests/unit/falcon_development/DevelopmentTest.php]: @spleshka we can use the same Drupal & DB modules with all kinds of tests! :)

File: Makefile [hunk 7]
```diff
@@ -1,5 +1,5 @@
 .PHONY: default pull up stop down clean exec exec-wodby exec-root drush \
-prepare prepare-composer prepare-files prepare-settings \
+prepare prepare-composer prepare-files prepare-settings test \
```
Comment (duozersk) [inline Makefile]: Could you please add a list of added commands here?

File: docs/testing.rst [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+- Run tests.
+
+If you are interested in details of Falcon CI setup,  you can find latest Circle CI configuration file here: https://github.com/systemseed/falcon/blob/master/.circleci/config.yml
+
+Running tests on local
+----------------------
+
+First, run ``make tests-prepare`` to initialize testing framework.
+Now, you can run all Codeception tests using ``make tests-run`` command. You can
+run a specific test suite by passing its name a second parameter: ::
```
Comment (duozersk) [inline docs/testing.rst]: "its name a second parameter" - shouldn't it be "its name as a second parameter"?

File: docs/testing.rst [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+It will enable autocomplete of available test methods in most of popular IDEs.
+Now you can start writing tests and run them using ``make tests-run``.
+
+To access ``codecept`` cli directly run: ::
+
+  make tests-cli
+
+It will allow you to run codecept commands to generate new tests or run a specific
+test if you need. See list available commands here: https://codeception.com/docs/reference/Commands
```
Comment (duozersk) [inline docs/testing.rst]: "See list available commands" -> "See list of available commands"

**Review by duozersk — CHANGES_REQUESTED** [review]

**Review by duozersk — APPROVED** [review]

**Pull request merged**